core/templates/pages/collection-player-page/collection-player-page.component.ts

```diff
@@ -0,0 +1,433 @@
+        countMiddleIcon = 1;
+      } else if (countMiddleIcon === 1 && x === this.ICON_X_MIDDLE_PX) {
+        x = this.ICON_X_RIGHT_PX;
+        y += this.ICON_Y_INCREMENT_PX;
+        countMiddleIcon = 0;
+      } else {
+        x = this.ICON_X_MIDDLE_PX;
+        y += this.ICON_Y_INCREMENT_PX;
+      }
+      iconParametersArray.push({
+        thumbnailIconUrl:
+          collectionNodes[i].getExplorationSummaryObject(
+          ).thumbnail_icon_url.replace(
+            'subjects', 'inverted_subjects'),
+        left: x + 'px',
+        top: y + 'px',
+        thumbnailBgColor:
+          collectionNodes[i].getExplorationSummaryObject(
+          ).thumbnail_bg_color
+      });
+    }
+    return iconParametersArray;
+  }
+
+  getExplorationUrl(explorationId: string): string {
+    return (
+      '/explore/' + explorationId + '?collection_id=' +
+      this.collectionId);
+  }
+
+  getExplorationTitlePosition(index: number): string {
+    if (index % 2 === 0) {
+      return '8px';
+    } else if ((index + 1) % 2 === 0 && (index + 1) % 4 !== 0) {
+      return '30px';
+    } else if ((index + 1) % 4 === 0) {
+      return '-40px';
+    }
+  }
+
+  scrollToLocation(id: string): void {
+    this.windowRef.nativeWindow.location.hash = '#' + (id);
+  }
+
+  closeOnClickingOutside(): void {
+    this.explorationCardIsShown = false;
+  }
+
+  onClickStopPropagation($evt: Event): void {
+    $evt.stopPropagation();
+  }
+
+  isCompletedExploration(explorationId: string): boolean {
+    if (this.collectionPlaythrough) {
+      let completedExplorationIds = (
+        this.collectionPlaythrough.getCompletedExplorationIds());
+      return completedExplorationIds.indexOf(explorationId) > -1;
+    }
+    return false;
+  }
+
+  ngOnInit(): void {
+    // $scope.$watch('$ctrl.collection', function(newValue) {
+    //   if (
+    //     newValue !== null &&
+    //     ctrl.collection.getCollectionNodeCount()) {
+    //     ctrl.generatePathParameters();
+    //   }
+    // }, true);
+    this.loaderService.showLoadingScreen('Loading');
+    this.collection = null;
+    this.collectionId = this.urlService.getCollectionIdFromUrl();
+    this.explorationCardIsShown = false;
+    // The pathIconParameters is an array containing the co-ordinates,
+    // background color and icon url for the icons generated on the
+    // path.
+    this.pathIconParameters = [];
+    this.activeHighlightedIconIndex = -1;
+    this.MIN_HEIGHT_FOR_PATH_SVG_PX = 220;
+    this.ODD_SVG_HEIGHT_OFFSET_PX = 150;
+    this.EVEN_SVG_HEIGHT_OFFSET_PX = 280;
+    this.ICON_Y_INITIAL_PX = 35;
+    this.ICON_Y_INCREMENT_PX = 110;
+    this.ICON_X_MIDDLE_PX = 225;
+    this.ICON_X_LEFT_PX = 55;
+    this.ICON_X_RIGHT_PX = 395;
+    this.svgHeight = this.MIN_HEIGHT_FOR_PATH_SVG_PX;
+    this.nextExplorationId = null;
+    this.whitelistedCollectionIdsForGuestProgress = (
+      AppConstants.WHITELISTED_COLLECTION_IDS_FOR_SAVING_GUEST_PROGRESS);
+    // $anchorScroll.yOffset = -80;
```

Comment:
```javascript
this.windowRef.nativeWindow.scrollTo()
or
this.windowRef.nativeWindow.scrollY
```
can be a substitute.

Reply:
Not sure if this works same as `this.windowRef.nativeWindow.scrollTo()`.  We've a different option for anchor scrolling in angular 2+ . Take a look at [this](https://medium.com/lacolaco-blog/introduce-router-scroller-in-angular-v6-1-ef34278461e9) blog post. But this needs to be enabled application wide.